lance-lakekeeper-test/test_lance_lakekeeper.py

```
@@ -0,0 +1,1168 @@
+            )
+            print(f"  Users data:\n{df.to_string(index=False)}")
+        elif name == "products":
+            check("B: products row count", len(df) == 3, f"got {len(df)}")
+            check("B: products has Widget", "Widget" in df["product_name"].values)
+            emb = df["embedding"].iloc[0]
+            check(
+                "B: embedding is list of 8 floats",
+                isinstance(emb, (list, np.ndarray)) and len(emb) == 8,
+                f"type={type(emb)}, len={len(emb) if hasattr(emb, '__len__') else 'N/A'}",
+            )
+            print(f"  Products data (without embeddings):")
```

Ruff (0.15.6)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)